src/common-helpers.ts

```diff
@@ -75,13 +75,26 @@ export const getMessageFromError = (error: Error): string => {
   return error.message;
 };
 
+/** Takes the original unvalidated examples from the properties of ZodObject schema shape */
+export const pullExampleProps = <T extends z.SomeZodObject>(subject: T) =>
+  Object.entries(subject.shape).reduce<Partial<z.input<T>>[]>(
+    (acc, [key, schema]) =>
+      combinations(
+        acc,
+        (schema._def[metaSymbol]?.examples || []).map(objOf(key)),
+        ([left, right]) => ({ ...left, ...right }),
+      ),
+    [],
+  );
+
 export const getExamples = <
   T extends z.ZodTypeAny,
   V extends "original" | "parsed" | undefined,
 >({
   schema,
   variant = "original",
   validate = variant === "parsed",
+  pullProps = true,
```

Comment:
perhaps this should not be default value.
can I find exact places where it should be activated?